### PR TITLE
Attempt to fix discussion deserialization errors

### PR DESCRIPTION
**Pull request by gdpelican**

### CODE REVIEW

**Comment by gdpelican** [comment]
Think this will do the trick @robguthrie ?

File: app/models/discussion.rb [hunk 1]
```diff
@@ -95,10 +95,14 @@ def thread_item_created!
...
     update_sequence_info!
-    RecalculateReadItemsCountsJob.perform_later(discussion)
-    true
+    update_reader_items_count!
   end
 
+  def update_reader_items_count!
+    discussion_readers.map(&:sync_read_items_count!)
+  end
+  handle_asynchronously :update_reader_items_count!, priority: 0
```
Comment (robguthrie) [inline app/models/discussion.rb]: default priority is 10, maybe 5? emails should be 100 or so.

File: app/models/discussion_reader.rb [hunk 2]
```diff
@@ -67,6 +67,10 @@ def discussion_reader_volume
     self.class.volumes.invert[self[:volume]]
   end
 
+  def sync_read_items_count!
+    update_attribute(:read_items_count, calculate_read_items_count)
```
Comment (robguthrie) [inline app/models/discussion_reader.rb]: I can't see any place in the server side where we use this anymore.

The client calculates it's own read items count pretty quickly anyway. I think we can drop the job ( and column later, with the other columns I'm yet to drop).

File: app/models/discussion_reader.rb [hunk 3]
```diff
@@ -88,15 +92,12 @@ def unread_ranges
...
 
   def read_ranges_string
-    if self[:read_ranges_string] == nil
+    self[:read_ranges_string] ||= begin
       if last_read_sequence_id == 0
         ""
       else
-        first = (discussion.first_sequence_id == 0) ? 1 : discussion.first_sequence_id
-        "#{first}-#{self.last_read_sequence_id}"
+        "#{[discussion.first_sequence_id, 1].max}-#{last_read_sequence_id}"
```
Comment (robguthrie) [inline app/models/discussion_reader.rb]: yep, looks correct to me.

**Comment by robguthrie** [comment]
thank you!

**Pull request merged**